### DISCUSSION TITLE
Check Tabular Model Status

### DISCUSSION THREAD

**Discussion by muhssamy** [TabularEditor/TabularEditor3#1368] [🙋♀️TE3 Q&A] [answered]
is there any C# script or any way to identify if the data model status is invalid? i need it as a Boolean either valid or not.

**Comment by otykier** [comment, marked as answer]
Could you clarify what you mean by "data model status is invalid"?

Many objects in the model have a `State` property. It is simple to check if those objects are all in the `Ready` state:

```csharp
var relationshipsReady = Model.Relationships.All(r => r.State == ObjectState.Ready);
var partitionsReady = Model.AllPartitions.All(r => r.State == ObjectState.Ready);
var measuresReady = Model.AllMeasures.All(r => r.State == ObjectState.Ready);
var columnsReady = Model.AllColumns.All(r => r.State == ObjectState.Ready);
var hierarchiesReady = Model.AllHierarchies.All(r => r.State == ObjectState.Ready);
var calculationItemsReady = Model.AllCalculationItems.All(r => r.State == ObjectState.Ready);

var modelReady = relationshipsReady && partitionsReady && measuresReady && columnsReady && hierarchiesReady && calculationItemsReady;

if(modelReady) Info("All objects are ready!");
else Info("Some objects are NOT ready!");
```

However, a model can be queried even if some objects are not ready - that all depends on the query.

**Reply by muhssamy** [reply]
Yes. thanks @otykier this is what I need. 
I thought there is a model state property. but this will definitely help me.

> However, a model can be queried even if some objects are not ready - that all depends on the query.

yes, but my issue if the data is cleared from the model by any how and there are some partitions not processed. the relationships also are not ready this leads to aggregate miss, and it cause a huge load to the server.

so, I am thinking of creating some validation in dev and UAT environment for any change before deploying to production. 
I have no clear process now but firstly I was checking if this validation is available and if you have a suggestion for me this will help.
